launch_testing/launch_testing/asserts/assert_output.py

```diff
@@ -45,7 +45,7 @@ def _match(expected, actual):
         return lambda expected, actual: expected in actual
     elif hasattr(expected_output, 'search'):
         return lambda expected, actual: (
-            expected.match(actual.replace(os.linesep, '\n')) is not None
+            expected.search(actual.replace(os.linesep, '\n')) is not None
```

Comment:
I encountered this same issue while trying to improve the test coverage on launch_testing.  A simple test that catches this issue and prevents regression would be something like

```
def test_works_with_regex(self):                 
    assertInStdout(                              
        self.proc_output,                        
        re.compile(r"Called with arguments \S+"),
        "terminating_proc-2"                     
    )                                            
```

You can add it to the bottom of `test_io_handler_and_assertions.py`  You'll also need to `import re` at the top

Reply:
Sounds good!